File: BOJ/#6603 로또/로또.py
from itertools import combinations

while True:
    nums = list(map(int, input().split()))
    k = nums[0]
    nums = nums[1:]
    if k == 0: break

    cmb = list(combinations(nums, 6))
    for cm in cmb:
        for c in cm: print(c, end=' ')
        print()

    print()

File: BOJ/#6603 로또/로또.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    while(1) {
        int k; cin>>k;
        if(k == 0) break;

        vector<int> nums(k);
        for(int i=0; i<k; cin>>nums[i++]);

        vector<bool> cmb(k, false);
        for(int i=0; i<6; i++) cmb[i] = true;

        vector<bool> v = cmb;
        while(1) {
            for(int i=0; i<k; i++)
                if(v[i]) printf("%d ", nums[i]);
            cout<<'\n';

            prev_permutation(v.begin(), v.end());
            if(v == cmb) break;
        }
        cout<<'\n';
    }
}
